corehq/pillows/domain.py

```diff
@@ -15,12 +15,24 @@
 
 def transform_domain_for_elasticsearch(doc_dict):
     doc_ret = copy.deepcopy(doc_dict)
-    sub = Subscription.visible_objects.filter(subscriber__domain=doc_dict['name'], is_active=True)
+    sub = (
+        Subscription
+            .visible_and_suppressed_objects
+            .filter(subscriber__domain=doc_dict['name'])
+            .filter(is_hidden_to_ops=False)
+            .filter(is_active=True)
+            .first()
+    )
     doc_ret['deployment'] = doc_ret.get('deployment', None) or {}
     countries = doc_ret['deployment'].get('countries', [])
     doc_ret['deployment']['countries'] = []
     if sub:
-        doc_ret['subscription'] = sub[0].plan_version.plan.edition
+        try:
+            doc_ret['subscription'] = sub.plan_version.plan.edition
+        except (SoftwarePlan.DoesNotExist, SoftwarePlan.MultipleObjectsReturned, AttributeError):
+            # AttributeError: 'Subscription' object has no attribute '_plan_version_cache'
+            # AttributeError: 'SoftwarePlanVersion' object has no attribute '_plan_cache'
```

Comment:
seems dangerous to swallow these as they surely indicate that something is broken?

Reply:
If I paste this into a Python console, it works. (I only have one domain locally.) I don't understand why I'm sometimes getting `DoesNotExist` or `MultipleObjectsReturned` exceptions.